muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/package.json

```diff
@@ -5,12 +5,12 @@
   "main": "index.frontpage.ts",
   "scripts": {
     "build": "webpack",
+    "build-full": "npm run build-client && npm run build",
     "test": "echo \"Error: no test specified\" && exit 1",
     "eslint": "eslint . --ext .ts,.tsx",
     "stylelint": "stylelint \"**/*.scss\"",
     "prettier-format": "prettier --config .prettierrc.json \"**/*.{ts,tsx}\" --write",
-    "build-client": "./node_modules/@openapitools/openapi-generator-cli/bin/openapi-generator generate -i ../../swagger.yaml -o generated/client -c generator-config.json -g typescript-fetch",
-    "build-client-windows": "openapi-generator-cli generate -i ../../swagger.yaml -o generated/client -c generator-config.json -g typescript-fetch"
+    "build-client": "openapi-generator-cli generate -i ../../swagger.yaml -o generated/client -c generator-config.json -g typescript-fetch"
```

Comment:
`npm run build-full` calls `build-client` script which seems to output `generated/client` folder even with full maven build as it is called in `pom.xml` right after `npm ci` is completed. 

Is this the correct way to build something to production as it seems that generated folder is included in the prod. build even if it was meant to be used in development only.

Previously we used `NODE_:ENV` variable to control how webpack made the build.

Reply:
You need that client to build frontend because frontend has references to generated client code "aka" interfaces and fetch methods. Without those frontend is not completed and thus cannot be build properly. 

What is not necessary is that generated/client is present in github, because we have swagger.yaml that holds every data to generate client when ever needed. This means that only when you are developing or building production you need to generated that client.

Why is pom.xml changed to use `npm run build-full`, because so that `mvn clean compile` which also builds frontend needs to have client present for the reasons previously mentioned. This should also simplify how backend developers can get frontend running without running multiple different commands separately on the frontend side. For frontend developers you can still run `mvn clean compile` once and from that point just run `npm run build` whenever you have someother frontend changes.